pkg/supportbundle/spec.go

```diff
@@ -251,6 +264,120 @@ func injectDefaults(app apptypes.AppType, b *troubleshootv1beta2.SupportBundle,
 	return supportBundle, nil
 }
 
+func addDiscoveredSpecs(
+	supportBundle *troubleshootv1beta2.SupportBundle, app apptypes.AppType,
+) *troubleshootv1beta2.SupportBundle {
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		logger.Errorf("Failed to get kubernetes clientset: %v", err)
+		return supportBundle
+	}
+
+	specs, err := findSupportBundleSpecs(clientset)
+	if err != nil {
+		logger.Errorf("Failed to find support bundle secrets: %v", err)
+		return supportBundle
+	}
+
+	for _, specData := range specs {
+		sbObject, err := sb.ParseSupportBundleFromDoc([]byte(specData))
+		if err != nil {
+			logger.Errorf("Failed to unmarshal support bundle spec: %v", err)
+			continue
+		}
+
+		// TODO: How about other spec fields e.g URI?
+		supportBundle.Spec.Collectors = append(supportBundle.Spec.Collectors, sbObject.Spec.Collectors...)
+		supportBundle.Spec.Analyzers = append(supportBundle.Spec.Analyzers, sbObject.Spec.Analyzers...)
+	}
+
+	return supportBundle
+}
+
+// findSupportBundleSpecs finds all support bundle secrets/configmaps in the cluster
+// The function will query all objects with troubleshoot.io/kind=support-bundle label
+// and, in code, filter out all kotsadm objects that have an object name
+// following kotsadm-<app-slug>-supportbundle format.
+// Reference: https://troubleshoot.sh/docs/support-bundle/discover-cluster-specs/
+func findSupportBundleSpecs(client kubernetes.Interface) ([]string, error) {
+	labelSelector := kotstypes.TroubleshootKey + "=" + kotstypes.TroubleshootValue
+	ctx := context.TODO()
+
+	specs := []string{}
+
+	// Get all namespaces
+	nsList, err := client.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Could listing all namespaces fail in a namespace-scoped installation?  Found some existing logic for gathering namespaces here that seems to take that into account: https://github.com/replicatedhq/kots/blob/main/pkg/helm/cache.go#L46-L57